### PR TITLE
DSP-11343: Protect against case where sc.defaultParallelism < numberNodes, avoid…

### PR DESCRIPTION
**Pull request by rocco408**
… coresPerNode = 0

### CODE REVIEW

**Comment by rocco408** [comment]
Avoid this problem: 
```
BASH:: ./run-test.sh dse -o 500000000 -y 10000 -p 10000 -r 1 jwcrpallcolumns --conf spark.dse.continuousPagingEnabled=true
About to Start Trials
ERROR 2017-02-07 18:55:40,785 org.apache.spark.deploy.DseSparkSubmitBootstrapper: Failed to start or submit Spark application
java.lang.IllegalArgumentException: requirement failed: size=0 and step=0, but both must be positive
	at scala.collection.Iterator$GroupedIterator.<init>(Iterator.scala:1023) ~[scala-library-2.11.8.jar:na]
	at scala.collection.Iterator$class.grouped(Iterator.scala:1160) ~[scala-library-2.11.8.jar:na]
	at scala.collection.AbstractIterator.grouped(Iterator.scala:1336) ~[scala-library-2.11.8.jar:na]
	at scala.collection.IterableLike$class.grouped(IterableLike.scala:174) ~[scala-library-2.11.8.jar:na]
	at scala.collection.AbstractIterable.grouped(Iterable.scala:54) ~[scala-library-2.11.8.jar:na]
```

File: src/main/scala/com/datastax/sparkstress/ReadTask.scala [hunk 1]
```diff
@@ -33,8 +33,10 @@ abstract class ReadTask(config: Config, sc: SparkContext) extends StressTask {
 
   val numberNodes = CassandraConnector(sc.getConf).withClusterDo( _.getMetadata.getAllHosts.size)
   val tenthKeys:Int = config.numTotalKeys.toInt / 10
-  val coresPerNode:Int = sc.defaultParallelism / numberNodes
 
+  val cores = if (sys.env.get("SPARK_WORKER_CORES") == None) numberNodes else (sys.env.get("SPARK_WORKER_CORES").get.toInt * numberNodes)
```
Comment (RussellSpitzer) [inline src/main/scala/com/datastax/sparkstress/ReadTask.scala]: more scalay to say
```sys.env.getOrElse("SPARK_WORKER_CORES", "1").toInt * numberNodes```

File: src/main/scala/com/datastax/sparkstress/ReadTask.scala [hunk 2]
```diff
@@ -33,8 +33,10 @@ abstract class ReadTask(config: Config, sc: SparkContext) extends StressTask {
...
+  val defaultParallelism = if (sc.defaultParallelism < numberNodes) cores else sc.defaultParallelism
```
Comment (RussellSpitzer) [inline src/main/scala/com/datastax/sparkstress/ReadTask.scala]: `math.max(sc.defaultParallelism, cores)`

**Review by RussellSpitzer — COMMENTED** [review]
few little style comments ( I do wonder if perhaps the root of the parallelism being the wrong number is a timing issue, this value probably changes as the executors register...)

**Pull request merged**